modules/igv/1.0/igv.smk

```diff
@@ -0,0 +1,824 @@
+            merged_lines = len(f.readlines())
+
+        with open(output_file, "a") as handle:
+            for line in batch_script:
+                if merged_lines > 0:
+                    if line.startswith(("load", "maxPanelHeight", "genome","setSleepInterval", "collapse")):
+                        continue
+                    if line.startswith(tuple(params.igv_options)) and not line.startswith("sort"):
+                        continue
+                handle.write(line)
+
+        batch_script.close()
+
+        output_touch = open(output.dispatched_batch_script, "w")
+        output_touch.close()
+
+def _evaluate_batches(wildcards):
+    CFG = config["lcr-modules"]["igv"]
+    checkpoint_output = checkpoints._igv_create_batch_script_per_variant.get(**wildcards).output.variant_batch
+
+    if wildcards.sample_id in list(CFG["runs"]["tumour_sample_id"]):
+        this_sample = op.filter_samples(CFG["runs"], tumour_sample_id=wildcards.sample_id, tumour_seq_type = wildcards.seq_type, tumour_genome_build = wildcards.genome_build)
+        normal_sample_id = this_sample["normal_sample_id"]
+        pair_status = this_sample["pair_status"]
+
+        maf = expand(
+            str(rules._igv_filter_maf.output.maf),
+            seq_type = wildcards.seq_type,
+            genome_build = wildcards.genome_build,
+            tumour_id = wildcards.sample_id,
+            normal_sample_id = normal_sample_id,
+            pair_status = pair_status
+        )
+
+        maf_table = pd.read_table(maf[0], comment = "#", sep="\t")
+
+    if wildcards.sample_id in list(CFG["runs"]["normal_sample_id"]):
+        these_samples = op.filter_samples(CFG["runs"], normal_sample_id=wildcards.sample_id, tumour_seq_type = wildcards.seq_type, tumour_genome_build = wildcards.genome_build, pair_status="matched") # only get MAFs from matched tumour + normal combos
+
+        mafs = expand(
+                str(rules._igv_filter_maf.output.maf),
+                zip,
+                seq_type = these_samples["tumour_seq_type"],
+                genome_build = these_samples["tumour_genome_build"],
+                tumour_id = these_samples["tumour_sample_id"],
+                normal_sample_id = these_samples["normal_sample_id"],
+                pair_status = these_samples["pair_status"]
+            )
+
+        maf_table = pd.concat([pd.read_table(m, comment="#", sep="\t") for m in mafs])
+
+    return expand(
+            expand(
+                str(rules._igv_batches_to_merge.output.dispatched_batch_script),
+                zip,
+                chromosome = maf_table["chr_std"],
+                start_position = maf_table["Start_Position"],
+                gene = maf_table["Hugo_Symbol"],
+                allow_missing = True
+            ),
+            preset = wildcards.preset,
+            allow_missing = True
+        )
+
+rule _igv_download_igv:
+    output:
+        igv_zip = CFG["dirs"]["igv"] + "IGV_2.7.2.zip",
```

Comment:
What if someone doesn't run this on linux or wants to have a different version? Maybe this part should be configurable or better even use the conda instead of downloading the source file?
https://anaconda.org/bioconda/igv